### ISSUE TITLE
Implementation with Alamofire

### ISSUE THREAD

**Issue by kimdv** [p2/OAuth2#59]
Hello! 

Is there a plan to implement p2.oauth2 in Alamofire in someway ? 

I think it could be nice, so you don't need to set the tokens when you make a request.

**Comment by p2** [comment]
No plans at this time. You can use [`oauth.request(forURL: NSURL)`](http://p2.github.io/OAuth2/Classes/OAuth2.html#/s:FC6OAuth26OAuth27requestFS0_FT6forURLCSo5NSURL_CS_13OAuth2Request) to get a NSURLRequest that is already signed.

**Comment by kimdv** [comment]
But i was thinking of automating this process

**Comment by p2** [comment]
Sure, I'll keep this open. But I'm not using Alamofire, so I'm unlikely to build this link any time soon.

**Comment by p2** [comment]
Maybe that sounds too negative, let me try again: Alamofire *may* be used for requests in the future, but I don't know how soon this will be.

**Comment by kimdv** [comment]
:+1:

**Comment by p2** [comment]
So one way to use OAuth2 with Alamofire is by setting the header manually:

```swift
let headers = [
    "Authorization": "Bearer \(oauth2.accessToken)",
]

Alamofire.request(.GET, "http://httpbin.org/get", headers: headers)
```

Alternatively, here's an extension so you can use your `OAuth2` instance in place of `Alamofire`:

```swift
import Alamofire

extension OAuth2 {
    public func request(
        method: Alamofire.Method,
        _ URLString: URLStringConvertible,
        parameters: [String: AnyObject]? = nil,
        encoding: Alamofire.ParameterEncoding = .URL,
        headers: [String: String]? = nil)
        -> Alamofire.Request
    {
        var hdrs = headers
        hdrs["Authorization"] = "Bearer \(accessToken)"
        return Alamofire.request(
            method,
            URLString,
            parameters: parameters,
            encoding: encoding,
            headers: hdrs)
    }
}

oauth2.request(.GET, "http://httpbin.org/get")
```

I have not tested this myself yet, feedback and suggestions welcome!

**Comment by p2** [comment]
Added to the readme on develop branch.
